tensorflow-framework/src/main/java/org/tensorflow/framework/initializers/TruncatedNormal.java

```diff
@@ -74,19 +73,11 @@ public TruncatedNormal(Ops tf, double mean, double stddev, long seed) {
 
   /** {@inheritDoc} */
   @Override
-  public <T extends TType> Operand<T> call(Operand<TInt64> dims, Class<T> type) {
-    if (!TNumber.class.isAssignableFrom(type)) {
-      throw new IllegalArgumentException("Tensor type must be numeric: " + type.getSimpleName());
-    }
-    @SuppressWarnings("unchecked")
-    Class<TNumber> nType = (Class<TNumber>) type;
+  public <U extends TFloating> Operand<U> call(Operand<TInt64> dims, Class<U> type) {
     long[] seeds = {seed, 0};
-    Operand<TNumber> distOp = tf.random.statelessTruncatedNormal(dims, tf.constant(seeds), nType);
-    return cast(
-        tf,
-        tf.math.add(
-            tf.math.mul(distOp, cast(tf, tf.constant(stddev), distOp.type())),
-            cast(tf, tf.constant(mean), distOp.type())),
-        type);
+    Operand<U> distOp = tf.random.statelessTruncatedNormal(dims, tf.constant(seeds), type);
+    return tf.math.add(
+        tf.math.mul(distOp, cast(tf, tf.constant(stddev), distOp.type())),
```

Comment:
The code shape here is different to `RandomNormal`. Can they be the same but only call a different `tf.random` function?